include/internal/catch_generators.hpp

```diff
@@ -349,6 +349,53 @@ namespace Generators {
         );
     }
 
+    template <typename T>
+    class RangeGenerator final : public IGenerator<T> {
+        T m_current;
+        T m_end;
+        T m_step;
+        bool m_positive;
+    public:
+        RangeGenerator(T const& start, T const& end, T const& step):
+            m_current(start),
+            m_end(end),
+            m_step(step),
+            m_positive(m_step > T(0))
+        {
+            assert(m_step != T(0) && "Step size cannot be zero");
+            assert((m_positive && m_current <= m_end) || (!m_positive && m_current >= m_end) && "Step moves away from end");
```

Comment:
you might want to enclose the or condition in parentheses so the message is reported for both checks. If i read it correctly now it should be reported only when the second condition triggers.